debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbIncrementalSnapshotContext.java

```diff
@@ -0,0 +1,289 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.mongodb;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Queue;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.DebeziumException;
+import io.debezium.annotation.NotThreadSafe;
+import io.debezium.pipeline.source.snapshot.incremental.IncrementalSnapshotContext;
+import io.debezium.relational.Table;
+import io.debezium.util.HexConverter;
+
+/**
+ * A class describing current state of incremental snapshot of MongoDB connector
+ *
+ * @author Jiri Pechanec
+ *
+ */
+@NotThreadSafe
+public class MongoDbIncrementalSnapshotContext<T> implements IncrementalSnapshotContext<T> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MongoDbIncrementalSnapshotContext.class);
+
+    // TODO Consider which (if any) information should be exposed in source info
```

Comment:
Same as above; all pending TODOs should be referenced in a follow-up Jira issue.

Reply:
Handled in follow-up Jira